### PR TITLE
[Pipeline] Fix incorrect argument order in the pipeline module

### PR DESCRIPTION
**Pull request by chhzh123**
## Description ##
This PR fixes the incorrect argument order in the pipeline module. See the following description. The previous implementation only fixes the argument order between pipeline stages, but does not handle the input mismatch.

```python
# GPT2LMHeadModel argument order
(input_ids, past_key_values, attention_mask, token_type_ids, position_ids, ...)
# Actual input arguments to the pipeline module
(input_ids, attention_mask, position_ids)
# Arguments of generated fx pipeline module 
(input_ids, position_ids, attention_mask)
```

## Checklist ##

- [x] PR's title starts with a category (e.g. [Bugfix], [Model], [Tutorial], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

cc @comaniac @zarzen

### CODE REVIEW

File: tests/test_pipeline_partition.py [hunk 1]
```diff
@@ -129,4 +231,5 @@ def forward(self, x):
 
 
 if __name__ == "__main__":
-    pytest.main([__file__])
+    # pytest.main([__file__])
+    test_pipeline_partition()
```
Comment (comaniac) [inline tests/test_pipeline_partition.py:235]: Mischange?

**Review by comaniac — APPROVED** [review]
Otherwise LGTM

**Comment by zarzen** [comment]
What will be the args list for passing from stage to stage?

**Comment by chhzh123** [comment]
> What will be the args list for passing from stage to stage?

`(hidden_state, position_ids, attention_mask)`, where `hidden_state` is the output of first stage, and `postiton_ids` and `attention_mask` are bypassed from the input arguments.

**Comment by zarzen** [comment]
> 

you mean the values `(position_ids, attention_mask)` are passed to the next stage but will be ignored by the second stage?

**Comment by chhzh123** [comment]
I mean `attention_mask` is required by each stage, so it should also be passed from the first stage to the second one.

I added two GPT test cases. You can check its expected output to see how the data are transferred between stages.
https://github.com/awslabs/slapo/blob/95f4b667226e5e00088bb7751c1a3fa4e5558de9/tests/test_pipeline_partition.py#L258-L277

**Pull request merged**